### DISCUSSION TITLE
`BaseGame` naming

### DISCUSSION THREAD

**Discussion by spydon** [flame-engine/flame#948] [Ideas]
Since `Game` and `BaseGame` will be components (after #906) and `Game` now has some more features than it used to have @st-pasha  suggested that its maybe time to change the name of `BaseGame` to something more representative.

**Comment by spydon** [comment]
@flame-engine/flame-colab @st-pasha for visibility

**Comment by st-pasha** [comment]
My suggestion was `FlameGame`, similar to how we now have `OxygenGame`

**Reply by spydon** [reply]
It is now named `FlameGame` (and `Game` is a mixin instead of a class) after #906 was merged.
![game_lowlevel(1)](https://user-images.githubusercontent.com/744771/133405893-6cb8bffd-9a27-42c8-8cb5-497f889ee57a.png)

**Comment by erickzanardo** [comment]
Before, Game and BaseGame differ as BaseGame as a component based game, now both are components, what are their difference now? Asking so I can think better about this haha

**Reply by spydon** [reply]
`BaseGame` is still a bit more opionated, it has:
 - the `Camera`
 - you can handle priority changes for the components in there
 - prepares the components by checking a lot of asserts and resizes it for the first time
 
 Maybe they could be merged to one, I'm not sure if that would cause any problems for Oxygen.

**Reply by wolfenrain** [reply]
I am mostly concerned that it will add more code at runtime when "everything" moves to `Game`, especially if some people don't use it. Not sure how the Dart tree shaking handles methods on class instances that are not used.

But apart from that, anything that "forces" people to use FCS and related functionality will be breaking for Oxygen I guess, so to prevent that I would opt for having it still separate where possible :-)

Another good reason to keep it separated would be what @spydon said about BaseGame, it is opinionated and that has the pro that people can still do other "fancy" stuff on `Game` itself.
